spec/services/measurements_creator_spec.rb

```diff
@@ -0,0 +1,148 @@
+        session = create_session!(type: "MoblieSession")
+        stream = create_stream!(session: session)
+
+        MeasurementsCreator.call(stream, single_measurement_attributes)
+
+        expect(Measurement.first.arrival_utc_time).to be_nil
+      end
+    end
+
+    context "when there are more than one measurements" do
+      it "schedules the creation of measurements" do
+        session = create_session!
+        stream = create_stream!(session: session)
+
+        MeasurementsCreator.call(stream, multiple_measurements_attributes)
+
+        assert_equal 1, AsyncMeasurementsCreator.jobs.size
+      end
+    end
+  end
+
+  describe "#call" do
+    let(:streams_repository) { double(StreamsRepository.new) }
+    let(:measurements_creator) { MeasurementsCreator.new(streams_repository) }
+    let(:stream) { double(Stream.new) }
+
+    before() do
+      allow(stream).to receive(:after_measurements_created)
+      allow(stream).to receive(:build_measurements!)
+      allow(streams_repository).to receive(:calc_average_value!)
+      allow(streams_repository).to receive(:calc_bounding_box!)
+      allow(stream).to receive(:fixed?)
+    end
+
+    context "for all sessions" do
+      it "delegates building the measurements" do
+        expect(stream).to receive(:build_measurements!).with(single_measurement_attributes)
+
+        measurements_creator.call(stream, single_measurement_attributes)
+      end
+
+      it "performs a callback after measuremnets are created" do
+        expect(stream).to receive(:after_measurements_created).with(no_args)
+
+        measurements_creator.call(stream, single_measurement_attributes)
+      end
+    end
+
+    context "for sessions that are not fixed" do
+      before() do
+        allow(stream).to receive(:fixed?).and_return(false)
+      end
+
+      it "calculates bounding box of stream" do
+        expect(streams_repository).to receive(:calc_bounding_box!).with(stream)
+
+        measurements_creator.call(stream, single_measurement_attributes)
+      end
+
+      it "calculattes average value of stream" do
+        expect(streams_repository).to receive(:calc_average_value!).with(stream)
+
+        measurements_creator.call(stream, single_measurement_attributes)
+      end
+    end
+  end
+
+  private
+
+  def multiple_measurements_attributes
+    [measurement_attributes, measurement_attributes]
+  end
+
+  def single_measurement_attributes
+    [measurement_attributes]
+  end
+
+  def measurement_attributes
+    { longitude: 25.4356212,
+      latitude: 56.4523456,
+      time: "2016-05-11T17:09:02",
+      milliseconds: 925,
+      measured_value: 59.15683475380729,
+      value: 59.15683475380729,
+    }
+  end
+
+  def create_stream!(attributes)
+    Stream.create!(
+      sensor_package_name: "AirBeam2:00189610719F",
+      sensor_name: "AirBeam2-F",
+      measurement_type: "Temperature",
+      unit_name: "Fahrenheit",
+      session: attributes[:session],
```

Comment:
If you are always expecting the key / value to be there `#fetch` is a better way. It raises if the key is missing. A fast failure can save hours of debugging.